### PR TITLE
chore(gateway): log timeouts as warn

### LINKED ISSUE #5397: Lower log level of connect exceptions in the gateway
**Description**

Connection exceptions are bound to happen in a distributed system - reporting on them as errors every time they occur isn't particularly useful, and is more useful as debugging information to diagnose what is going on. At the moment, in the gateway, we will get the following log: "Expected to handle gRPC request, but an unexpected error occurred", with the stack trace:

```
io.netty.channel.ConnectTimeoutException: connection timed out: zeebe-2.zeebe-broker-service.2f0480c8-5d8a-4d3b-861b-178f6f7a36b3-zeebe.svc.cluster.local/10.4.11.168:26501
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:575) ~[netty-transport-native-epoll-4.1.50.Final-linux-x86_64.jar:4.1.50.Final]
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98) ~[netty-common-4.1.50.Final.jar:4.1.50.Final]
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:170) ~[netty-common-4.1.50.Final.jar:4.1.50.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) ~[netty-common-4.1.50.Final.jar:4.1.50.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) ~[netty-common-4.1.50.Final.jar:4.1.50.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384) ~[netty-transport-native-epoll-4.1.50.Final-linux-x86_64.jar:4.1.50.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.50.Final.jar:4.1.50.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.50.Final.jar:4.1.50.Final]
	at java.lang.Thread.run(Unknown Source) ~[?:?]
```

I would prefer to log these as a debugging information, as it's only useful to figure out what's going on if anything goes wrong - however a case can be made for WARN, as such errors may cause oddities to occur (even if they are recovered).

EDIT: This also occurs when the connection is refused (though it will be retried again - see https://console.cloud.google.com/errors/COLP4ZfE4drbdQ?service=zeebe&version=0.24.2&time=P1D&project=camunda-cloud-240911&organizationId=669107107215)

### PR DESCRIPTION
**Pull request by sccalabr**
…gger

## Description

The log level was lowered from error to warn for connection time out exceptions.

## Related issues

closes #5397

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by jwulf** [comment]
Thanks for the contribution @sccalabr. Looks like you nailed the commit message - well done. I've requested a review from @npepinpe.

File: gateway/src/main/java/io/zeebe/gateway/EndpointManager.java [hunk 1]
```diff
@@ -419,7 +420,12 @@ public static StatusRuntimeException convertThrowable(final Throwable cause) {
       status = Status.NOT_FOUND.augmentDescription(cause.getMessage());
       Loggers.GATEWAY_LOGGER.debug(
           "Expected to handle gRPC request, but request could not be delivered", cause);
-    } else {
+    } else if (cause instanceof ConnectTimeoutException) {
+      status = status.augmentDescription(cause.getMessage());
+      Loggers.GATEWAY_LOGGER.debug(
```
Comment (Zelldon) [inline gateway/src/main/java/io/zeebe/gateway/EndpointManager.java:425]:
```suggestion
      Loggers.GATEWAY_LOGGER.warn(
```

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Hey @sccalabr thanks for your contribution :tada: 

I discussed that with @npepinpe and we would like to have the logging level as warn instead of debug could you please change that then we can merge it.

Greets
Chris

File: gateway/src/main/java/io/zeebe/gateway/EndpointManager.java (hunk 1 above)
Comment (sccalabr) [reply gateway/src/main/java/io/zeebe/gateway/EndpointManager.java]: Done :)

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Hey @sccalabr 

thanks for the change. Sorry I missed it on the first review you commit message need to be changed as well.

**TL;DR**

Current:
```
fix(broker): Connection Timeouts Logged at Debug Level for Gateway Lo… …
…gger
```

Ideally it would be:
```
chore(gateway): log timeouts as warn
```

**Description:**

Your PR is not really a bug fix, which means we should not use `fix` as type here. We use `chore` for these kind of changes.
The scope should normally be the maven module, which was under change the most. In your PR it is `gateway` not `broker`.
The message of your commit is to long, which means it will be shorten by github. Ideally it contains only 72 characters take a look at this guide for more information https://www.robertcooper.me/git-commit-messages#subject-line-less-than-or-equal-to-72-characters-in-length

In general you can also read our Contribution guidelines for more information
https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines

Greets
Chris

**Comment by Zelldon** [comment]
Hey @sccalabr 

thanks for applying my review hints!
Unfortunately there are some merge conflicts. Could you please rebase to develop so we can merge your PR. If you need help with that let me know.

Greets
Chris

**Comment by sccalabr** [comment]
Let me know if I did it right.

**Comment by Zelldon** [comment]
@sccalabr the issue is that we moved most of the stuff into a new class called `GrpcErrorMapper` I think you need to make your changes there.

**Comment by sccalabr** [comment]
Would I just add the logging statement or something else

**Comment by sccalabr** [comment]
Looks like I somehow closed it? Not sure how to reopen
